src/main/java/com/github/harti2006/neo4j/StartNeo4jServerMojo.java

```diff
@@ -120,8 +151,49 @@ private void startNeo4jServer() throws MojoExecutionException {
             } else {
                 throw new MojoExecutionException("Neo4j server did not start up properly");
             }
+            
+            // Now we need to wait it replies
+            checkServerReady ();
+            
+            // Went up! If it's a new DB, let's change the password.
+            setNewPassword ();
         } catch (IOException | InterruptedException e) {
-            throw new MojoExecutionException("Could not start neo4j server", e);
+            throw new MojoExecutionException( "Could not start neo4j server", e);
         }
     }
+    
+    private void checkServerReady () throws MojoExecutionException, InterruptedException
+    {
+    		String pwd = deleteDb ? "neo4j" : password;
+    		int maxAttempts = 3;
+    		Thread.sleep ( 1500 ); // It takes some time anyway
+    		
+      for ( int attempts = 1; attempts <= maxAttempts ; attempts++ )
+      {
+      		try 
+      		{
+      			getLog().debug ( "Trying to connect Neo4j, attempt " + attempts ); 
+	        Driver driver = GraphDatabase.driver( "bolt://127.0.0.1:" + boltPort, AuthTokens.basic ( "neo4j", pwd ) );
+	      		driver.close ();
```

Comment:
Let's make use of the `AutoClosable` (try-with-resources) feature, e.g.
```java
try (Driver ignored = GraphDatabase.driver("bolt://127.0.0.1:" + boltPort,
                                           AuthTokens.basic("neo4j", pwd))) {
    return;
} catch (ServiceUnavailableException ignored) {
    Thread.sleep(3000);
}
```

Reply:
WOW! I didn't know Java got this, thanks! I'll fix it this way.